api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,199 @@
+
+                    if (dvlaVehicle.getMotTestExpiryDate() != null) {
+                        return Response.ok(buildMotResponse(dvlaVehicle)).build();
+                    }
+
+                    throw new InvalidResourceException("No HGV/PSV vehicle retrieved for registration " + registration,
+                            awsRequestId);
+                }
+
+                HgvPsvVehicle hgvPsvVehicle = buildHgvPsvResponse(hgvPsvVehicleOptional.get(), dvlaVehicle);
+
+                return Response.ok(hgvPsvVehicle).build();
+            } else {
+                logger.error("No MOT Test or DVLA vehicle found for registration.");
+                throw new InvalidResourceException("No MOT Test or DVLA vehicle found for registration " + registration,
+                        awsRequestId);
+            }
+        } catch (TradeException e) {
+            // no need to log these errors, just throw them back
+            throw e;
+        } catch (Exception e) {
+            // log all unhandled exceptions and throw an internal server error
+            logger.error(e);
+            throw new InternalServerErrorException(e, awsRequestId);
+        } finally {
+            logger.trace("Exiting getVehicle");
+        }
+    }
+
+    private Optional<MotrResponse> getMotVehicle(String registration) throws Exception {
+        logger.info(String.format("Retrieving vehicle by registration: %s", registration));
+
+        MotrResponse motVehicles;
+
+        try {
+            motVehicles = motrReadService.getLatestMotTestByRegistration(registration);
+        } catch (Exception e) {
+            logger.error(String.format("There was an error retrieving mot vehicle with reg: %s",
+                    registration), e);
+            throw new InvalidResourceException("There was an error retrieving mot vehicle", awsRequestId);
+        }
+
+        return Optional.ofNullable(motVehicles);
+    }
+
+    private Optional<MotrResponse> getDvlaVehicle(String registration) throws Exception {
+        logger.info(String.format("Retrieving dvla vehicle by registration: %s", registration));
+
+        MotrResponse dvlaVehicle;
+
+        try {
+            dvlaVehicle = motrReadService.getLatestMotTestForDvlaVehicleByRegistration(registration);
+        } catch (Exception e) {
+            logger.error(String.format("There was an error retrieving dvla vehicle with reg: %s",
+                    registration), e);
+            throw new InvalidResourceException("There was an error retrieving dvla vehicle", awsRequestId);
+        }
+
+        return Optional.ofNullable(dvlaVehicle);
+    }
+
+    private Optional<Vehicle> getHgvPsvVehicle(String registration) throws Exception {
+        Vehicle foundVehicle = null;
+
+        try {
+            if (!Strings.isNullOrEmpty(registration)) {
+                foundVehicle = hgvVehicleProvider.getVehicle(registration);
+            } else {
+                logger.warn("Passed registration was empty or null");
```

Comment:
is it possible to pass an empty registration ?

Reply:
not possible from the integrated motr->trade perspective. Also not possible for HGV/PSV only. Might become possible with trailers when calling api with empty string directly? Currently this would be useful only in the off-chance that there is a dvla_vehicle with registration field empty, which is possible (from DB schema perspective). 

I'd leave that in since its only two lines and has some theoretical value